category_encoders/quantile_encoder.py

```diff
@@ -0,0 +1,291 @@
+"""Target Encoder"""
+__author__ = "david26694", "cmougan"
+
+import numpy as np
+import pandas as pd
+from sklearn.base import BaseEstimator
+from category_encoders.ordinal import OrdinalEncoder
+from category_encoders.m_estimate import MEstimateEncoder
+import category_encoders.utils as util
+from sklearn.utils.random import check_random_state
+class QuantileEncoder(BaseEstimator, util.TransformerWithTargetMixin):
+    """Quantile Encoding for categorical features.
+    This a statistically modified version of target MEstimate encoder where selected features
+    are replaced the statistical quantile instead than the mean. Replacing with the
+    median is a particular case where self.quantile = 0.5. In comparison to MEstimateEncoder
+    it has two tunable parameter `m` and `quantile`
+    Parameters
+    ----------
+    verbose: int
+        integer indicating verbosity of the output. 0 for none.
+    quantile: int
+        integer indicating statistical quantile. ´0.5´ for median.
+    m: int
+        integer indicating the smoothing parameter. 0 for no smoothing.
+    cols: list
+        a list of columns to encode, if None, all string columns will be encoded.
+    drop_invariant: bool
+        boolean for whether or not to drop columns with 0 variance.
+    return_df: bool
+        boolean for whether to return a pandas DataFrame from transform (otherwise it will be a numpy array).
+    handle_missing: str
+        options are 'error', 'return_nan'  and 'value', defaults to 'value', which returns the target quantile.
+    handle_unknown: str
+        options are 'error', 'return_nan' and 'value', defaults to 'value', which returns the target quantile.
+    Example
+    -------
+    >>> from category_encoders import *
+    >>> import pandas as pd
+    >>> from sklearn.datasets import load_boston
+    >>> bunch = load_boston()
+    >>> y = bunch.target
+    >>> X = pd.DataFrame(bunch.data, columns=bunch.feature_names)
+    >>> enc = QuantileEncoder(cols=['CHAS', 'RAD']).fit(X, y)
```

Comment:
as a user i think it would be nice to have the tunable parameters mentioned here like 
```enc = QuantileEncoder(cols=['CHAS', 'RAD'], quantile=0.5, m=1.0).fit(X, y)```